test/e2e/disruption.go

```diff
@@ -42,63 +42,123 @@ var _ = framework.KubeDescribe("DisruptionController [Feature:PodDisruptionbudge
 	})
 
 	It("should create a PodDisruptionBudget", func() {
-		pdb := policy.PodDisruptionBudget{
+		createPodDisruptionBudgetOrDie(cs, ns, intstr.FromString("1%"))
+	})
+
+	It("should update PodDisruptionBudget status", func() {
+		createPodDisruptionBudgetOrDie(cs, ns, intstr.FromInt(2))
+
+		createPodsOrDie(cs, ns, 2)
+
+		err := wait.PollImmediate(framework.Poll, 60*time.Second, func() (bool, error) {
+			pdb, err := cs.Policy().PodDisruptionBudgets(ns).Get("foo")
+			if err != nil {
+				return false, err
+			}
+			return pdb.Status.PodDisruptionAllowed, nil
```

Comment:
Because I'm too lazy to check the code: am I correct in remembering that PodDisruptionAllowed starts out false, therefore this really only flips to true when the controller has updated it after it has seen the two pods? (Obviously if it starts out true, this doesn't test anything. I'm sure you thought of this, just wanted to make sure.)